### ISSUE TITLE
Add new collection protocol to a Donor

### ISSUE THREAD

**Issue by anthonyhorlock** [AberystwythSystemsBiology/LImBuS#217]
I feel that it would be useful to be able to add a collection protocol to a Donor (e.g., for longitudinal, cohort, cross-sectional collections...).

![image](https://user-images.githubusercontent.com/98022410/201337260-29c65924-2faf-4d55-92b8-ffcc8448b36d.png)

It would then be useful to have an additional filter in the Donor Index to be able to search according to 'Collection':

![image](https://user-images.githubusercontent.com/98022410/201345137-8fd7e06b-b4dc-4b79-a949-a674fa4adc28.png)

**Comment by pipaLU** [comment]
Sth to be discussed, currently the collection is sample based. Perhaps, better to allow flexible functions for tracing the samples linked to the consents for studies.

**Comment by anthonyhorlock** [comment]
Thanks Chuan! I think I could only assign a Collection to a sample as an 'Other Protocol Event'

![image](https://user-images.githubusercontent.com/98022410/201357600-27c27ea0-626c-4623-9607-731e983281bc.png)

However, after I assigned this to a sample, it didn't come up when I filtered for it in the sample index. 

It has taken me a while to figure out which index (Sample or Donor) would be the best to be able to identify what samples we could provide from the Biobank. I believe that the Donor Index is the most suitable for this purpose.
